src/backend/vulkan/src/device.rs

```diff
@@ -666,32 +605,103 @@ impl d::Device<B> for Device {
+                        }
+                    }
+                    let colors = subpass.colors.iter().map(make_ref).collect::<Box<[_]>>();
+                    let depth_stencil = subpass.depth_stencil.map(make_ref);
+                    let inputs = subpass.inputs.iter().map(make_ref).collect::<Box<[_]>>();
+                    let preserves = subpass
+                        .preserves
+                        .iter()
+                        .map(|&id| id as u32)
+                        .collect::<Box<[_]>>();
+                    let resolves = subpass.resolves.iter().map(make_ref).collect::<Box<[_]>>();
+
+                    (colors, depth_stencil, inputs, preserves, resolves)
+                })
+                .collect::<Box<[_]>>();
+
+            let subpasses = attachment_refs
+                .iter()
+                .map(|(colors, depth_stencil, inputs, preserves, resolves)| {
+                    vk::SubpassDescription {
+                        flags: vk::SubpassDescriptionFlags::empty(),
+                        pipeline_bind_point: vk::PipelineBindPoint::GRAPHICS,
+                        input_attachment_count: inputs.len() as u32,
+                        p_input_attachments: inputs.as_ptr(),
+                        color_attachment_count: colors.len() as u32,
+                        p_color_attachments: colors.as_ptr(),
+                        p_resolve_attachments: if resolves.is_empty() {
+                            ptr::null()
+                        } else {
+                            resolves.as_ptr()
+                        },
+                        p_depth_stencil_attachment: match depth_stencil {
+                            Some(ref aref) => aref as *const _,
+                            None => ptr::null(),
+                        },
+                        preserve_attachment_count: preserves.len() as u32,
+                        p_preserve_attachments: preserves.as_ptr(),
+                    }
+                })
+                .collect::<Box<[_]>>();
+
+            inplace_it::inplace_or_alloc_array(dependencies.len(), |uninit_guard| {
+                let dependencies =
+                    uninit_guard.init_with_iter(dependencies);
+
+                let info = vk::RenderPassCreateInfo {
+                    s_type: vk::StructureType::RENDER_PASS_CREATE_INFO,
+                    p_next: ptr::null(),
+                    flags: vk::RenderPassCreateFlags::empty(),
+                    attachment_count: attachments.len() as u32,
+                    p_attachments: attachments.as_ptr(),
+                    subpass_count: subpasses.len() as u32,
+                    p_subpasses: subpasses.as_ptr(),
+                    dependency_count: dependencies.len() as u32,
+                    p_dependencies: dependencies.as_ptr(),
+                };
+
+                let result = self.shared.raw.create_render_pass(&info, None);
+
+                match result {
```

Comment:
let's try to move things out of `inplace_or_alloc_array` where possible. In this case, for example, we should do
```rust
let result = inplace_it::inplace_or_alloc_array(dependencies.len(), |uninit_guard| {
  ..
});
match result {
  ...
}
```